### ISSUE TITLE
configuration parameters in command line

### ISSUE THREAD

**Issue by jesseholzerpnnl** [GOCompetition/C3DataUtilities#5]
Is anybody using the “-c” or “--configuration” argument of the check_data.py script in C3DataUtilities? If not, I’d like to repurpose it as an output file, rather than an input file. Currently it is an input file for values of configuration parameters. But the "--parameters" argument allows you to override the defaults. Given that, I would like to output the final parameter values, after overrides are applied, and it would be natural to supply the name of this output file as the “-c” or “--configuration” argument. If this change works for everybody, the interface would be:
* default values of parameters are in a configuration file that should not be changed
* default values can be overridden by supplying a JSON string in the "--parameters" argument
*  final values after overrides are written to a configuration file whose name is supplied in the "--configuration" argument
I'm especially interested in feedback (pro or con) on this from @arunveeramany, @ccoffrin, @Robbybp, @tarekelgindy, @scottgreene, @blue442, @fsafdarian, @wmakGIT, and any others using check_data.py in data preparation.

**Comment by wmakGIT** [comment]
For me, I don't use the -c or the --configuration option. If I need to change any options, I usually just change the default values in the default configuration file though.

**Comment by jesseholzerpnnl** [comment]
> For me, I don't use the -c or the --configuration option. If I need to change any options, I usually just change the default values in the default configuration file though.

Yes, you can still do that. Just don't commit any changes to config.json

**Comment by Robbybp** [comment]
I haven't used the -c option yet, but I would recommend adding a new argument to write the final config parameters rather than changing the behavior of an existing argument. Maybe something like --final-config or --write-config-to.

**Comment by jesseholzerpnnl** [comment]
> I haven't used the -c option yet, but I would recommend adding a new argument to write the final config parameters rather than changing the behavior of an existing argument. Maybe something like --final-config or --write-config-to.

good point.

**Comment by ccoffrin** [comment]
If not one is really using the -c option so far, I would not be too concerned about changing its semantics at this point.  That said, I do agree human readable argument names like @Robbybp suggests are a good thing.
